Core/src/org/sleuthkit/autopsy/timeline/actions/AddManualEvent.java

```diff
@@ -0,0 +1,299 @@
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME;
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DESCRIPTION;
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_TL_EVENT_TYPE;
+import org.sleuthkit.datamodel.DataSource;
+import org.sleuthkit.datamodel.SleuthkitCase;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.timeline.EventType;
+
+/**
+ * Action that allows the user the manually create timeline events. It prompts
+ * the user for event data and then adds it to the case via an artifact.
+ */
+@NbBundle.Messages({
+    "AddManualEvent.text=Add Event",
+    "AddManualEvent.longText=Manually add an event to the timeline."})
+public class AddManualEvent extends Action {
+
+    private final static Logger logger = Logger.getLogger(AddManualEvent.class.getName());
+    private static final String MANUAL_CREATION = "Manual Creation";
+    private static final Image ADD_EVENT_IMAGE = new Image("/org/sleuthkit/autopsy/timeline/images/add.png", 16, 16, true, true, true); // NON-NLS
+
+    private final TimeLineController controller;
+
+    /**
+     * Initialize the custom value extractor used by the ValidationSupport for
+     * the LocalDateTimeTextField in the EventCreationDialogPane.
+     */
+    static {
+        ValueExtractor.addObservableValueExtractor(LocalDateTimeTextField.class::isInstance,
+                control -> ((LocalDateTimeTextField) control).localDateTimeProperty());
+    }
+
+    /**
+     * Create an Action that allows the user the manually create timeline
+     * events. It prompts the user for event data with a dialog and then adds it
+     * to the case via an artifact. The datetiem in the dialog will be set to
+     * "now" when the action is invoked.
+     *
+     * @param controller The controller for this action to use.
+     *
+     */
+    public AddManualEvent(TimeLineController controller) {
+        this(controller, null);
+    }
+
+    /**
+     * Create an Action that allows the user the manually create timeline
+     * events. It prompts the user for event data with a dialog and then adds it
+     * to the case via an artifact.
+     *
+     * @param controller  The controller for this action to use.
+     * @param epochMillis The initial datetime to populate the dialog with. The
+     *                    user can ove ride this.
+     */
+    public AddManualEvent(TimeLineController controller, Long epochMillis) {
+        super(Bundle.AddManualEvent_text());
+        this.controller = controller;
+        setGraphic(new ImageView(ADD_EVENT_IMAGE));
+        setLongText(Bundle.AddManualEvent_longText());
+
+        setEventHandler(actionEvent -> {
+            //shoe the dialog and if it completed normally add the event.
+            new EventCreationDialog(controller, epochMillis).showAndWait().ifPresent(this::addEvent);
```

Comment:
This might be an issue in EventCreationDialog, but when the dialog pops up its not confined to the bounds of the main window.  In the particular case of my setup the dialog pops up on my other screen, far away from the main window.   This also means that I can put the main window on top of the popup dialog thus hiding the the popup causing the main window to appear "stuck".

Reply:
I also noticed that I was allowed to popup more than one add event dialog.